### PR TITLE
Updates FHiCLs for CAF to read weights from SBNEventWeight moduls in sbncode

### PR DESCRIPTION
**Pull request by Ooohu**
Update `cafmaker_defs.fcl`: 

Change "eventweight_genie3_sbn_A.fcl" to "eventweight_genie_sbn.fcl";
Add "eventweight_flux_sbn.fcl"

### CODE REVIEW

File: fcl/caf/cafmakerjob_icarus_sce_genie_and_fluxwgt.fcl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #include "cafmakerjob_icarus_sce.fcl"
 
-services.RandomNumberGenerator: {}
+physics.runprod: [ @sequence::caf_ana_sce_evtw_sequence, cafmaker, genieweight, fluxweight]
```
Comment (cjbackhouse) [inline fcl/caf/cafmakerjob_icarus_sce_genie_and_fluxwgt.fcl:3]:
```suggestion
physics.runprod: [ @sequence::caf_ana_sce_evtw_sequence, genieweight, fluxweight, cafmaker ]
```
Pretty sure causes need to precede effects :wink:

**Review by cjbackhouse — CHANGES_REQUESTED** [review]

**Review by cjbackhouse — APPROVED** [review]
Looks right to me, but we should check it actually runs.

**Comment by Ooohu** [comment]
Tested. It runs!

**Review by jzennamo — APPROVED** [review]

**Comment by jzennamo** [comment]
This addresses issue https://github.com/SBNSoftware/icaruscode/issues/250 and I think it is ready to go!

**Review by FernandaPsihas — APPROVED** [review]

**Comment by FernandaPsihas** [comment]
@SFBayLaser, would it be possible to merge this before the 2021C freeze this week? Thank you!

**Comment by SFBayLaser** [comment]
Yes, working on getting some stuff in for a release based on last week's LArSoft... at this rate hopefully tomorrow for that.

**Comment by SFBayLaser** [comment]
Hi Folks, I should have done the old "trigger build" before I merged this PR... this PR is causing the CI tests to fail during the check fhicl stage, you can see the problem at https://dbweb8.fnal.gov:8443/LarCI/app/ns:icarus/storage/docs/2021/11/17/testFHiCLfiles_icarus.sh.log
I don't want to back this PR out... so can someone please very quickly take a look at this and fix? 
Thanks!

**Comment by Ooohu** [comment]
> Hi Folks, I should have done the old "trigger build" before I merged this PR... this PR is causing the CI tests to fail during the check fhicl stage, you can see the problem at https://dbweb8.fnal.gov:8443/LarCI/app/ns:icarus/storage/docs/2021/11/17/testFHiCLfiles_icarus.sh.log I don't want to back this PR out... so can someone please very quickly take a look at this and fix? Thanks!

I fixed it here https://github.com/SBNSoftware/sbncode/pull/192#issuecomment-972546278
Basically I should have used the name `@local::sbn_eventweight_genie` all the time, but somehow I used `@local::sbn_genie_eventweight` (should have drop this) instead. That calls the variable defined here `sbncode/SBNEventWeight/jobs/genie/eventweight_genie_sbn.fcl`

**Pull request merged**